scripts/install-worker.sh

```diff
@@ -335,6 +367,75 @@ if [[ ! $KUBERNETES_VERSION =~ "1.19"* && ! $KUBERNETES_VERSION =~ "1.20"* && !
     sudo mv $TEMPLATE_DIR/ecr-credential-provider-config /etc/eks/ecr-credential-provider/ecr-credential-provider-config
 fi
 
+################################################################################
+### Cache Images ###############################################################
+################################################################################
+if [[ "$CACHE_CONTAINER_IMAGES" = "true" && "$BINARY_BUCKET_REGION" != "us-iso-east-1" && "$BINARY_BUCKET_REGION" != "us-isob-east-1" ]]; then
+    AWS_DOMAIN=$(imds 'latest/meta-data/services/domain')
+    ECR_URI=$(/etc/eks/get-ecr-uri.sh "${BINARY_BUCKET_REGION}" "${AWS_DOMAIN}")
+
+    PAUSE_CONTAINER="${ECR_URI}/eks/pause:${PAUSE_CONTAINER_VERSION}"
+    KUBE_PROXY_IMG="${ECR_URI}/eks/kube-proxy:v${KUBERNETES_VERSION}-${KUBE_PROXY_VERSION_SUFFIX}"
+    KUBE_PROXY_MINIMAL_IMG="${ECR_URI}/eks/kube-proxy:v${KUBERNETES_VERSION}-minimal-${KUBE_PROXY_VERSION_SUFFIX}"
+    cat /etc/eks/containerd/containerd-config.toml | sed s,SANDBOX_IMAGE,$PAUSE_CONTAINER,g | sudo tee /etc/eks/containerd/containerd-cached-pause-config.toml
+    sudo cp -v /etc/eks/containerd/containerd-cached-pause-config.toml /etc/containerd/config.toml
+
+    sudo cp -v /etc/eks/containerd/sandbox-image.service /etc/systemd/system/sandbox-image.service
+    sudo chown root:root /etc/systemd/system/sandbox-image.service
+    sudo systemctl daemon-reload
+    sudo systemctl start containerd
+    sudo systemctl enable containerd sandbox-image
+    /etc/eks/containerd/pull-image.sh "${PAUSE_CONTAINER}"
+    /etc/eks/containerd/pull-image.sh "${KUBE_PROXY_IMG}"
+    /etc/eks/containerd/pull-image.sh "${KUBE_PROXY_MINIMAL_IMG}"
```

Comment:
Tested the script with `make 1.23` and I got this error message.

```
2022-10-16T00:37:32+08:00:     amazon-ebs: ‘/etc/eks/containerd/containerd-cached-pause-config.toml’ -> ‘/etc/containerd/config.toml’
2022-10-16T00:37:32+08:00:     amazon-ebs: ‘/etc/eks/containerd/sandbox-image.service’ -> ‘/etc/systemd/system/sandbox-image.service’
2022-10-16T00:37:33+08:00:     amazon-ebs: Created symlink from /etc/systemd/system/multi-user.target.wants/containerd.service to /usr/lib/systemd/system/containerd.service.
2022-10-16T00:37:33+08:00:     amazon-ebs: Created symlink from /etc/systemd/system/multi-user.target.wants/sandbox-image.service to /etc/systemd/system/sandbox-image.service.
2022-10-16T00:37:33+08:00:     amazon-ebs:
2022-10-16T00:37:33+08:00:     amazon-ebs: Unable to locate credentials. You can configure credentials by running "aws configure".
2022-10-16T00:37:48+08:00:     amazon-ebs: Attempt 1 of 3
2022-10-16T00:37:49+08:00:     amazon-ebs:
2022-10-16T00:37:49+08:00:     amazon-ebs: Unable to locate credentials. You can configure credentials by running "aws configure".
2022-10-16T00:38:18+08:00:     amazon-ebs: Attempt 2 of 3
2022-10-16T00:38:19+08:00:     amazon-ebs:
2022-10-16T00:38:19+08:00:     amazon-ebs: Unable to locate credentials. You can configure credentials by running "aws configure".
2022-10-16T00:39:02+08:00:     amazon-ebs: Attempt 3 of 3
2022-10-16T00:39:02+08:00:     amazon-ebs:
2022-10-16T00:39:02+08:00:     amazon-ebs: Unable to locate credentials. You can configure credentials by running "aws configure".
2022-10-16T00:39:02+08:00:     amazon-ebs: /etc/eks/containerd/pull-image.sh: line 15: return: exit: numeric argument required

( hang here, must hit CTRL-C to exit )
```

Seems like there are two issues need to resolve,

1. Instance have no Instance Profile associated, so there would have no ECR permission.
2. `$rc` does not capture as expected in `pull-image.sh`.

Reply:
guarded against empty ecr_password